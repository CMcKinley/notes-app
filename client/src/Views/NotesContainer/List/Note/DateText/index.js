import React, { memo } from "react";
import PropTypes from "prop-types";
import moment from "moment";

const DateText = ({ dateToDisplay, text }) => {
  const formattedDate = dateToDisplay
    ? moment(dateToDisplay).format("MMMM, DD, YYYY")
    : null;
  const formattedTime = dateToDisplay
    ? moment(dateToDisplay).format("h:mma")
    : null;
  const today = moment().format("MMMM, DD, YYYY");

  return (
    <span>
      {text} {today === formattedDate ? "Today" : formattedDate} at{" "}
      {formattedTime}
    </span>
  );
};

DateText.propTypes = {
  text: PropTypes.string,
  dateToDisplay: PropTypes.string
};

DateText.defaultProps = {
  text: "",
  dateToDisplay: ""
};

export default memo(DateText);
