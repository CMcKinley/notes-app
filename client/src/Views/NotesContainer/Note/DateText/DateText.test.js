import React from "react";
import renderer from "react-test-renderer";
import DateText from "./index.js";

describe("DateText", () => {
  it("Renders correctly", () => {
    const component = renderer.create(<DateText dateToDisplay="2019-10-30T00:46:18.276Z"  />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
