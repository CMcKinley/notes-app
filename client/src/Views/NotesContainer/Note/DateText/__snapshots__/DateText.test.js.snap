// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateText Renders correctly 1`] = `
<span>
  
   
  Today
   at
   
  8:46pm
</span>
`;
